data/mods/gen7/abilities.ts

```diff
@@ -66,6 +66,15 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
 		inherit: true,
 		onBoost() {},
 	},
+	soundproof: {
+		inherit: true,
+		onTryHit(target, source, move) {
+			if (move.target !== 'self' && move.flags['sound']) {
```

Comment:
```suggestion
			if (move.flags['sound']) {
```
This was just a naive way to support Clangorous Soul before I made the connection with Howl and Perish Song. https://github.com/smogon/pokemon-showdown/commit/1082258f84e3bf26390279e0168670fc4b705314

Reply:
Fair enough... I've rebased the backout of that commit onto my branch to make it clearer.